sys/dev/isp/isp.c

```diff
@@ -432,11 +559,28 @@ isp_reset(ispsoftc_t *isp, int do_load_defaults)
 	MBSINIT(&mbs, MBOX_EXEC_FIRMWARE, MBLOGALL, 5000000);
 	mbs.param[1] = code_org >> 16;
 	mbs.param[2] = code_org;
-	if (!IS_26XX(isp))
-		mbs.param[3] = loaded_fw ? 0 : 1;
+	mbs.param[3] = loaded_fw ? 0 : 1;
```

Comment:
As I don't have any specs at all, my whole work is based on reading and understanding the Linux driver provided by Marvell/QLogic.
There is no differentiating for this register in the Linux driver. It's just `mcp->mb[3] = 0;` there, no matter if firmware was loaded or not and what adapter/chip generation.

Reply:
I haven't read the code closely to be sure loaded_fw is always set for the later HBAs.  If that is the case, it may be work OK and only slightly confusing logically.

Reply:
Until now, `loaded_fw` would only be set for **24xx and 25xx** when an ispfw(4) was actually loaded. In this case a `MBOX_VERIFY_CHECKSUM` was issued first and a `MBOX_EXEC_FIRMWARE` right after it with `mbs.param[3]` set based on the value of `loaded_fw`.
For 26xx+ there was no ispfw(4) firmware until now. so `loaded_fw` is always `0` and the code explicitely issued a `MBOX_LOAD_FLASH_FIRMWARE`. As `MBOX_VERIFY_CHECKSUM` is hidden behind a `if (loaded_fw)` this was not issued for 26xx+ and the `MBOX_EXEC_FIRMWARE` got issued without setting `mbs.param[3]` to anything at all.

Reply:
I changed this back to be spec conform.
I've tested this with and without the change and could not see any difference.